### PR TITLE
Fix introspection on PDO_pgsql with `PDO::STRINGIFY_FETCHES` enabled

### PR DESCRIPTION
**Pull request by keulinho**
|      Q       |   A
|------------- | -----------
| Type         | improvement

#### Summary

Add more unit test jobs that use the PDO::STRINGIFY_FETCHES option, as discussed in this PR: https://github.com/doctrine/dbal/pull/7028#pullrequestreview-3042233545

I didn't add a job for `pdo_oci` and `pdo_sqlsrv` as for those drivers the result in `QueryBuilderTest` was already stringified, therefore I assume those drivers always stringify the fetches, therefore a separate job is not needed

### CODE REVIEW

**Comment by keulinho** [comment]
@derrabus would love your review on this one as well ;) 

The postgres job already uncovered similar issues like the MySQL one we ran into

For the other tests I personally think it is ok to just skip them for that specific job as it seems they cover some edge cases anyway and only run for specific drivers & configs

File: tests/Functional/FetchBooleanTest.php [hunk 1]
```diff
@@ -6,13 +6,17 @@
...
 use PHPUnit\Framework\Attributes\DataProvider;
 
 class FetchBooleanTest extends FunctionalTestCase
 {
     protected function setUp(): void
     {
-        if ($this->connection->getDatabasePlatform() instanceof PostgreSQLPlatform) {
+        if (
+            $this->connection->getDatabasePlatform() instanceof PostgreSQLPlatform
+            && ! TestUtil::isPdoStringifyFetchesEnabled()
```
Comment (derrabus) [inline tests/Functional/FetchBooleanTest.php:18]: If we stringify everything, we obviously can't expect actual booleans being returned from the database. Makes sense. Below we emit a message on why we skip the test. The message is now kind of odd because we actually do skip the test on Postgres sometimes.

Can we update it accordingly, for the next person who stumbles over this skipped test?

File: tests/Functional/SQL/ParserTest.php [hunk 2]
```diff
@@ -11,8 +11,8 @@ class ParserTest extends FunctionalTestCase
 {
     public function testPostgreSQLJSONBQuestionOperator(): void
     {
-        if (! TestUtil::isDriverOneOf('pdo_pgsql')) {
-            self::markTestSkipped('This test requires the pdo_pgsql driver.');
+        if (! TestUtil::isDriverOneOf('pdo_pgsql') || TestUtil::isPdoStringifyFetchesEnabled()) {
+            self::markTestSkipped('This test requires the pdo_pgsql driver without PDO::STRINGIFY_FETCHES option.');
```
Comment (derrabus) [inline tests/Functional/SQL/ParserTest.php]: Can we duplicate the test for the stringify case? It should still work, but the returned value is probably different.

**Review by derrabus — CHANGES_REQUESTED** [review]
Thank you again! So the Postgres driver will also fail if we have that option enabled, am I right? We should treat this PR as a bugfix then.

File: tests/Functional/FetchBooleanTest.php (hunk 1 above)
Comment (keulinho) [reply tests/Functional/FetchBooleanTest.php:18]: updated 👍

**Comment by keulinho** [comment]
> So the Postgres driver will also fail if we have that option enabled, am I right?

@derrabus Yes exactly, IMHO this should be treated as a bugfix

**Review by derrabus — APPROVED** [review]
